domain/infosync/info.go

```diff
@@ -82,17 +83,30 @@ type ServerVersionInfo struct {
 	GitHash string `json:"git_hash"`
 }
 
-var globalInfoSyncer *InfoSyncer
+var globalInfoSyncer atomic.Value
+
+func getGlobalInfoSyncer() (*InfoSyncer, error) {
+	v := globalInfoSyncer.Load()
+	if v == nil {
+		return nil, errors.New("infoSyncer is not initialized")
+	}
+	return v.(*InfoSyncer), nil
+}
+
+func setGlobalInfoSyncer(is *InfoSyncer) {
+	globalInfoSyncer.Store(is)
+}
 
 // GlobalInfoSyncerInit return a new InfoSyncer. It is exported for testing.
 func GlobalInfoSyncerInit(ctx context.Context, id string, etcdCli *clientv3.Client) (*InfoSyncer, error) {
-	globalInfoSyncer = &InfoSyncer{
+	is := &InfoSyncer{
 		etcdCli:        etcdCli,
 		info:           getServerInfo(id),
 		serverInfoPath: fmt.Sprintf("%s/%s", ServerInformationPath, id),
 		minStartTSPath: fmt.Sprintf("%s/%s", ServerMinStartTSPath, id),
 	}
-	return globalInfoSyncer, globalInfoSyncer.init(ctx)
+	setGlobalInfoSyncer(is)
+	return is, is.init(ctx)
```

Comment:
Is it correct that `setGlobalInfoSyncer` first and then `init`?

Reply:
done.